stor/s3.py

```diff
@@ -585,16 +585,26 @@ def download(self, dest, condition=None, use_manifest=False, **kwargs):
                     executor.submit(download_w_config, file_to_download): file_to_download
                     for file_to_download in files_to_download
                 }
-                for fut in as_completed(futures):
-                    result = fut.result()
+                for fut in as_completed(futures.keys()):
+                    source, dest = futures[fut]
```

Comment:
is this a tuple? or is it a dict `{'source': file, 'dest': dest}`

Reply:
It should be a dict. Looks like I did not add that update.

Basically we won't have this line and the exception should call `f'{futures[fut]["source"]}: {e}'` directly instead.

Reply:
Fixed in [b496d1a](https://github.com/counsyl/stor/pull/144/commits/b496d1a326f921591a00cacfa79a4125c8224286)